### PR TITLE
Switch to Poetry, upgrade, lint and cleanup

### PR DESCRIPTION
**Pull request by abulte**
My new passion is switching projects to Poetry, so this is done.

While struggling with it, a few bonuses:
- upgrade pandas and pandas-profiling, since python 3.9 is now a requirement (dependencies were impossible to compute on python 3.7) — we'll have to switch the production env
- fix tests locally
- fix the CI that was apparently not running (wrong circle deploy key)
- upgrade testing tooling
- add a linting step (flake8) to CI because code quality was starting to get creepy 🧟 — I fixed the linting errors from `kafka-integration` 🎁

### CODE REVIEW

File: csvapi/profileview.py [hunk 1]
```diff
@@ -17,23 +17,26 @@
...
 
     def get_dataframe(self, db_info, dtype=None):
         dsn = 'file:{}?immutable=1'.format(db_info['db_path'])
-        conn = sqlite3.connect(dsn)
+        conn = sqlite3.connect(dsn, uri=True)
         sql = 'SELECT * FROM [{}]'.format(db_info['table_name'])
         try:
             df = pd.read_sql_query(sql, con=conn, dtype=dtype)
-        except:
+        # TODO: check if correct exception type
```
Comment (maudetes) [inline csvapi/profileview.py:30]: Could you check that it was the correct exception type?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 ## Current (in progress)
 
-- Nothing yet
+- [BREAKING] Migrate to python >= 3.9
```
Comment (maudetes) [inline CHANGELOG.md:5]: Seems to work with my 3.8.10 version. We should update the Readme that mentions Python 3.7+ in any case.

**Review by maudetes — COMMENTED** [review]
Nice!

I'm getting warnings now on `csvapi serve`. I didn't have these on `kafka-integration` branch, but it may be related to version pinned.

```
<frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 232 from PyObject
```

File: csvapi/profileview.py (hunk 1 above)
Comment (abulte) [reply csvapi/profileview.py:30]: Don't know how 🤷 Anyway this code does not look very clean and tested, introducing a breakage should not be the end of the world.

File: CHANGELOG.md (hunk 2 above)
Comment (abulte) [reply CHANGELOG.md:5]: It _might_ work but the dependencies are explicitly computed for 3.9. Agreed for the README!

File: csvapi/profileview.py (hunk 1 above)
Comment (maudetes) [reply csvapi/profileview.py:30]: I'm ok with that. Tagging @geoffreyaldebert here so that he can be aware of this.

**Pull request merged**